packages/vite/src/node/server/middlewares/transform.ts

```diff
@@ -93,21 +93,10 @@ export function transformMiddleware(
       return
     }
 
-    let url: string
-    try {
-      url = removeTimestampQuery(req.url!).replace(NULL_BYTE_PLACEHOLDER, '\0')
-    } catch (err) {
-      // if it starts with %PUBLIC%, someone's migrating from something
-      // like create-react-app
-      let errorMessage: string
-      if (req.url?.startsWith('/%PUBLIC')) {
-        errorMessage = `index.html shouldn't include environment variables like %PUBLIC_URL%, see https://vitejs.dev/guide/#index-html-and-project-root for more information`
-      } else {
-        errorMessage = `Vite encountered a suspiciously malformed request ${req.url}`
-      }
-      next(new Error(errorMessage))
-      return
-    }
+    let url = removeTimestampQuery(req.url!).replace(
+      NULL_BYTE_PLACEHOLDER,
+      '\0'
+    )
 
     const withoutQuery = cleanUrl(url)
```

Comment:
I also got a question about  `cleanUrl`. Does it also do the work that `removeTimestampQuery` does? If so, could I remove `removeTimestampQuery` function?